services/chainstorage/src/main/java/tech/pegasys/teku/services/chainstorage/EphemeryDatabaseReset.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.chainstorage;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
+import tech.pegasys.teku.service.serviceutils.ServiceConfig;
+import tech.pegasys.teku.storage.server.Database;
+import tech.pegasys.teku.storage.server.VersionedDatabaseFactory;
+
+public class EphemeryDatabaseReset {
+
+  /** This method is called only on Ephemery network when reset is due. */
+  Database resetDatabaseAndCreate(
+      final ServiceConfig serviceConfig, final VersionedDatabaseFactory dbFactory) {
+    try {
+      final Path beaconDataDir = serviceConfig.getDataDirLayout().getBeaconDataDirectory();
```

Comment:
we need to be careful here, we're actually deleting more than we want.
beacon has more than just the db folder in it...
All I want deleted is `db` and `network.yml` - the rest can stay, and some is troublesome to lose sometimes like `kvstore` folder.
We should verify this in a non mock test also, or expand mock tests to show that we're only calling this on a few specific paths.